### ISSUE TITLE
Batchable: return data is always parsed for error string

### ISSUE THREAD

**Issue by dmihal** [boringcrypto/BoringSolidity#3]
Batchable.sol has the following code:

```solidity
    (bool success, bytes memory result) = address(this).delegatecall(calls[i]);
    require(success || !revertOnFail, _getRevertMsg(result));
```

This will execute `_getRevertMsg` every time, even when the delegatecall is successful.

In the best case, this just wastes a little bit of gas. In the worst case, it can cause the whole transaction to fail with an "out of gas" error, if `_getRevertMsg` is unable to parse the return data.

This can be fixed by replacing the require statement with this:

```solidity
    if (!success && revertOnFail) {
        revert(_getRevertMsg(result));
    }
```

**Comment by Mansarde** [comment]
Looks like this has been fixed in c66550dad8fb6c5f2f1ba246bc5ecc1676ef67a4

**Comment by boringcrypto** [comment]
Yeah, great find! Thanks!
